lib/FrameManager.js

```diff
@@ -454,17 +451,29 @@ class Frame {
       document.head.appendChild(script);
       return script;
     }
+
+    /**
+     * @param {string} content
+     * @return {!Promise<!ElementHandle>}
+     */
+    async function evaluateScriptContent(content) {
+      content += '\nwindow.pptrCsp = true; //# Detect Content Security Policy';
+      const script = await context.evaluateHandle(addScriptContent, content);
+      const cspInactive = await context.evaluate(() => window['pptrCsp']);
+      if (!cspInactive) throw new Error('Refused to execute inline script because it violates the Content Security Policy');
```

Comment:
We should set `pptrCsp` to false to avoid confusing ourselves the second time through this function. I think it would also be safter is `pptrCsp` was replaced by a random string or a static counter.

Reply:
+1, I was thinking of a unique identifier as well